Capture match on feature detection example

mattlemx opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

I added parentheses to an example where a regex expression needed to capture the matched capital letter for use in the replacement value. 
For example, 
The current example returns:
str.replace(/[A-Z]/g, "z$1") // 'jz$1z$1hy'

But expects:
str.replace(/([A-Z])/g, "z$1") // 'jzQzWhy' 

### Motivation

To fix an error in the example code execution. 

### Additional details

None.

### Related issues and pull requests

mattlemx committed:
Capture match on feature detection example

The example, str.replace(/[A-Z]/g, "z$1").split(/z(?=[A-Z])/g), as written outputs "jz$1z$1hy" when provided, "jQWhy", because the capital letter isn't captured. This can be fixed by wrapping [A-Z] with parentheses, str.replace(/([A-Z])/g, "z$1").

Josh-Cena reviewed (dismissed):
Thank you!

Josh-Cena reviewed (commented):
Well actually, this doesn't work as expected—`splitUpString("fooBare")` outputs `["foo", "Bare"]` instead. Could you try to make the example work as stated?

mattlemx committed:
Merge branch 'main' into patch-1

mattlemx committed:
Fixed return value of splitUpString examples

The fallback code returns ['foo', 'Bar'] instead of the expected ['fooB', 'ar'] (Camel Case). I updated the example to capture and replace capital letters in the string with a period following each capital letter and then split the string by the period.

mattlemx committed:
Merge branch 'main' into patch-1

mattlemx commented:
@Josh-Cena Thank you for taking time to review! I updated the second part of the `splitUpString` and it now outputs the correct ["fooB", "ar"]. **Note: I also changed fooBare to fooBar which is why the e is missing** I updated the example to capture and replace capital letters in the string with a period following each capital letter and then split the string by the period. Let me know if there is anything else.

mattlemx committed:
Merge branch 'main' into patch-1

mattlemx commented:
@Josh-Cena @teoli2003 I'm following up to see if there is anything else I can contribute to this pull request. Have a great weekend!

mattlemx committed:
Merge branch 'main' into patch-1

Josh-Cena commented on files/en-us/web/http/browser_detection_using_the_user_agent/index.md:
```diff
@@ -47,10 +47,10 @@ if (navigator.userAgent.includes("Chrome")) {
 } else {
   // This fallback code is much less performant, but works
-  splitUpString = (str) => str.replace(/[A-Z]/g, "z$1").split(/z(?=[A-Z])/g);
+  splitUpString = (str) => str.replace(/([A-Z])/g, "$1.").split('.');
```
This would not work with `splitUpString("foo.bar")`, which should not be split. Frankly I don't know what this function is doing exactly anyway (it's not a typical camel-case-splitting function), but I would suggest something a bit more robust:

```js
splitUpString = (str) => {
  const segments = str.split(/([A-Z])/g);
  const res = [segments[0]];
  for (let i = 1; i < segments.length; i += 2) {
    res.push(segments[i] + segments[i + 1]);
  }
};
```

mattlemx replied:
@Josh-Cena You're right, the return value should be consistent regardless of which version the browser supports. 

I made a change that uses regex to split the string when it matches a 0 or more characters up to and including a capital letter. The result includes empty strings between each element in the array that I then filter out.

mattlemx committed:
Fixed fallback splitUpString function to split camel case when period is included in the string.

mattlemx committed:
Merge branch 'main' into patch-1

mattlemx commented on files/en-us/web/http/browser_detection_using_the_user_agent/index.md:
```diff
@@ -43,10 +43,10 @@ if (navigator.userAgent.includes("Chrome")) {
 } else {
   // This fallback code is much less performant, but works
-  splitUpString = (str) => str.replace(/[A-Z]/g, "z$1").split(/z(?=[A-Z])/g);
+  splitUpString = (str) => String(str).split(/(.*?[A-Z])/).filter(Boolean);
```
@Josh-Cena I made a change that uses regex to split the string when it matches a 0 or more characters up to and including a capital letter. The result includes empty strings between each element in the array that I then filter out.

mattlemx committed:
Merge branch 'main' into patch-1

Josh-Cena reviewed (approved):
Thanks, that looks very cool!

Josh-Cena commented:
@mattlemx Thank you for all the work to find a clever solution. Congratulations on your first commit here—welcome aboard!

schalkneethling commented:
Welcome to the community, @mattlemx  🎉 Thank you for your contribution. If you are interested in more ways to contribute to MDN Web Docs, please have a look at our [contributor task board](https://github.com/orgs/mdn/projects/25). If you have any questions or need help finding something to work on, please ask for help in our [community discussions](https://github.com/orgs/mdn/discussions). We look forward to more of your contributions. 🙌
